app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,43 @@
+class CommentsController < ApplicationController
+
+  def new
+    @picture = Picture.find(params[:picture_id])
+    @comment = Comment.new
+  end
+
+  def create
+    @picture = Picture.find(params[:picture_id])
+    @comment = @picture.build_comment(comment_params, current_user)
+    if @comment.save
+      redirect_to picture_path(@picture)
+    else
+      render 'new'
+    end
+  end
+
+  def edit
+    @comment = Comment.find(params[:id])
```

Comment:
Your comments should already have a link to Pictures, so you can just get it from there rather than making a second query